nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-5-processors/src/main/java/org/apache/nifi/processors/elasticsearch/JsonQueryElasticsearch5.java

```diff
@@ -0,0 +1,451 @@
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    static {
+        final Set<Relationship> _rels = new HashSet<>();
+        _rels.add(REL_SUCCESS);
+        _rels.add(REL_FAILURE);
+        _rels.add(REL_HITS);
+        _rels.add(REL_AGGREGATIONS);
+        relationships = Collections.unmodifiableSet(_rels);
+
+        final List<PropertyDescriptor> descriptors = new ArrayList<>();
+        descriptors.add(PROP_SSL_CONTEXT_SERVICE);
+        descriptors.add(CONNECT_TIMEOUT);
+        descriptors.add(SOCKET_TIMEOUT);
+        descriptors.add(RETRY_TIMEOUT);
+        descriptors.add(USERNAME);
+        descriptors.add(PASSWORD);
+        descriptors.add(HTTP_HOSTS);
+        descriptors.add(QUERY);
+        descriptors.add(INDEX);
+        descriptors.add(TYPE);
+        descriptors.add(BREAK_UP_HITS);
+        descriptors.add(BREAK_UP_AGGREGATIONS);
+
+        propertyDescriptors = Collections.unmodifiableList(descriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+
+    @OnUnscheduled
+    public void onUnscheduled() {
+        this.client = null;
+    }
+
+    private void setupClient(ProcessContext context) throws Exception {
+        final String hosts = context.getProperty(HTTP_HOSTS).evaluateAttributeExpressions().getValue();
+        String[] hostsSplit = hosts.split(",[\\s]*");
+        final SSLContextService sslService =
+                context.getProperty(PROP_SSL_CONTEXT_SERVICE).asControllerService(SSLContextService.class);
+        final String username = context.getProperty(USERNAME).evaluateAttributeExpressions().getValue();
+        final String password = context.getProperty(PASSWORD).evaluateAttributeExpressions().getValue();
+
+        final Integer connectTimeout = context.getProperty(CONNECT_TIMEOUT).asInteger();
+        final Integer readTimeout    = context.getProperty(SOCKET_TIMEOUT).asInteger();
+        final Integer retryTimeout   = context.getProperty(RETRY_TIMEOUT).asInteger();
+
+        HttpHost[] hh = new HttpHost[hostsSplit.length];
+        for (int x = 0; x < hh.length; x++) {
+            URL u = new URL(hostsSplit[x]);
+            hh[x] = new HttpHost(u.getHost(), u.getPort(), u.getProtocol());
+        }
+
+        RestClientBuilder builder = RestClient.builder(hh)
+        .setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
+            @Override
+            public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
+                if (sslService != null) {
+                    try {
+                        KeyStore keyStore = KeyStore.getInstance(sslService.getKeyStoreType());
+                        KeyStore trustStore = KeyStore.getInstance("JKS");
```

Comment:
I think we should be using sslService.getTrustStoreType() here, in case it is set to something other than JKS